### ISSUE TITLE
Cannot access directories beginning with # (number sign)

### ISSUE THREAD

**Issue by septs** [rclone/rclone#4184]
#### What is the problem you are having with rclone?
Cannot access directories beginning with number sign

#### What is your rclone version (output from `rclone version`)
```
rclone v1.51.0
- os/arch: linux/amd64
- go version: go1.13.7
```

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)
Ubuntu 18.04.4 LTS

####  Which cloud storage system are you using? (eg Google Drive)
OneDrive for Business

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)
```
boxadmin@PAR-L3426 ~ $ rclone copy '1drv:#internal/books' books -P
2020-04-29 05:43:07 ERROR : : error reading source directory: directory not found
2020-04-29 05:43:07 ERROR : Attempt 1/3 failed with 1 errors and: directory not found
2020-04-29 05:43:07 ERROR : : error reading source directory: directory not found
2020-04-29 05:43:07 ERROR : Attempt 2/3 failed with 1 errors and: directory not found
2020-04-29 05:43:07 ERROR : : error reading source directory: directory not found
2020-04-29 05:43:07 ERROR : Attempt 3/3 failed with 1 errors and: directory not found
Transferred:             0 / 0 Bytes, -, 0 Bytes/s, ETA -
Errors:                 1 (retrying may help)
Elapsed time:         0.0s
2020/04/29 05:43:07 Failed to copy: directory not found
```

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)
```
boxadmin@PAR-L3426 ~ $ rclone -vv copy '1drv:#internal/books' books -P
2020/04/29 05:44:16 DEBUG : rclone: Version "v1.51.0" starting with parameters ["rclone" "-vv" "copy" "1drv:#internal/books" "books" "-P"]
2020/04/29 05:44:16 DEBUG : Using config file from "/home/boxadmin/.config/rclone/rclone.conf"
2020-04-29 05:44:20 ERROR : : error reading source directory: directory not found
2020-04-29 05:44:20 INFO  : Local file system at /home/boxadmin/books: Waiting for checks to finish
2020-04-29 05:44:20 INFO  : Local file system at /home/boxadmin/books: Waiting for transfers to finish
2020-04-29 05:44:20 ERROR : Attempt 1/3 failed with 1 errors and: directory not found
2020-04-29 05:44:20 ERROR : : error reading source directory: directory not found
2020-04-29 05:44:20 INFO  : Local file system at /home/boxadmin/books: Waiting for checks to finish
2020-04-29 05:44:20 INFO  : Local file system at /home/boxadmin/books: Waiting for transfers to finish
2020-04-29 05:44:20 ERROR : Attempt 2/3 failed with 1 errors and: directory not found
2020-04-29 05:44:20 ERROR : : error reading source directory: directory not found
2020-04-29 05:44:20 INFO  : Local file system at /home/boxadmin/books: Waiting for checks to finish
2020-04-29 05:44:20 INFO  : Local file system at /home/boxadmin/books: Waiting for transfers to finish
2020-04-29 05:44:20 ERROR : Attempt 3/3 failed with 1 errors and: directory not found
Transferred:             0 / 0 Bytes, -, 0 Bytes/s, ETA -
Errors:                 1 (retrying may help)
Elapsed time:         0.0s
2020/04/29 05:44:20 Failed to copy: directory not found
```

--------

![image](https://user-images.githubusercontent.com/3842474/80559030-eb4ba580-8a0e-11ea-8b64-838cc518b2bb.png)

**Comment by septs** [comment]
Use `@` or `.` with starts, can access

**Comment by ncw** [comment]
Hmm, this looks like an encoding problem.

We normally encode `#` characters into a similar looking unicode `＃` character.

However it looks like onedrive **does** allow `#` in directory names.

Here is one work-around


```
$ rclone lsf  --onedrive-encoding Slash,LtGt,DoubleQuote,Colon,Question,Asterisk,Pipe,Percent,BackSlash,Del,Ctl,LeftSpace,LeftTilde,RightSpace,RightPeriod,InvalidUtf8,Dot onedrive:#test
logo_symbol_color_16px.png
```

Another would be to rename the `#` to `＃` then it will work as expected.

Rclone uses the superset of reserved characters for onedrive business and onedrive

https://docs.microsoft.com/en-us/onedrive/developer/rest-api/concepts/addressing-driveitems?view=odsp-graph-online#path-encoding

Though this doesn't agree with this document

https://support.office.com/en-us/article/Invalid-file-names-and-file-types-in-OneDrive-OneDrive-for-Business-and-SharePoint-64883a5d-228e-48f5-b3d2-eb39e07630fa#invalidcharacters

**Comment by septs** [comment]
@ncw what the current status?

**Comment by Cnly** [comment]
Hi @septs, this should have been fixed by #4909. A beta version has been uploaded here: https://beta.rclone.org/branch/onedrive-dont-encode-hash-percent/v1.54.0-beta.5038.2aed3d595.onedrive-dont-encode-hash-percent/. Maybe you could help test it if you've got some time? :)

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.54
